brats/transformations.py

```diff
@@ -251,3 +234,55 @@ def _(mask: torch.Tensor, classes: typing.List[int]) -> torch.Tensor:
     for class_id, label in enumerate(filter(lambda label: label != 0, classes)):
         transformed[class_id][mask[0, ...] == label] = 1
     return transformed
+
+
+class CommonTransformation(abc.ABC):
+    """
+    Interface for transformations that are supposed to transform multiple inputs in the same way.
+    """
+
+    @abc.abstractmethod
+    def __call__(self, imgs: typing.List[typing.Union[np.ndarray, torch.Tensor]]):
+        raise NotImplementedError
+
+
+class RandomCrop(CommonTransformation):
+    def __init__(self, size: typing.Tuple[int, int]):
+        self.size = size
+
+    def __call__(self, *imgs: typing.Union[np.ndarray, torch.Tensor]) -> typing.Tuple[
+        typing.Union[np.ndarray, torch.Tensor],
+        typing.Union[np.ndarray, torch.Tensor]]:
+        assert all(
+            img[0, 0, ...].shape == imgs[0][0, 0, ...].shape for img in imgs), "W and H of all images must be the same"
+
+        max_x = imgs[0].shape[2] - self.size[0]
+        max_y = imgs[0].shape[3] - self.size[1]
+        x, y = random.randint(0, max_x), random.randint(0, max_y)
+        transformed = []
+        for img in imgs:
+            transformed_img = utils.copy(img)
+            transformed_img = transformed_img[:, :, x:x + self.size[0], y:y + self.size[1]]
+            transformed.append(transformed_img)
+
+        return tuple(transformed)
+
+
+class ComposeCommon(CommonTransformation):
+    """
+    Compose multiple transformations into one
+    :param transform: List of transforms to perform.
+    """
+
+    def __init__(self, transforms: typing.List[CommonTransformation]):
+        self.transforms = transforms
+
+    def __call__(self, *imgs: typing.Union[np.ndarray, torch.Tensor]) -> typing.Tuple[
+        typing.Union[np.ndarray, torch.Tensor],
+        typing.Union[np.ndarray, torch.Tensor]]:
+        assert all(
+            img[0, 0, ...].shape == imgs[0][0, 0, ...].shape for img in imgs), "W and H of all images must be the same"
+        transformed = [utils.copy(img) for img in imgs]
```

Comment:
You first copy here, and then copy in call to RandomCrop. Is this much copying really necessary? Why is it necessary at all?

Reply:
I would like every entry to be individual entity, so the variables would not be changed by accident (by changing other variable referencing the same object).